torch/quantization/_equalize.py

```diff
@@ -0,0 +1,120 @@
+import torch
+import copy
+
+def channel_range(input, axis=0):
+    ''' finds the range of weights associated with a specific channel
+
+    '''
+    size_of_tensor_dim = len(input.size())
+    mins = input
+    maxs = input
+
+    # reshape input to make specified axis the last axis in input
+    temp = list(range(size_of_tensor_dim))
+    temp[size_of_tensor_dim - 1] = axis
+    temp[axis] = size_of_tensor_dim - 1
+
+    mins = mins.permute(temp)
+    maxs = maxs.permute(temp)
+
+    # minimizing over all axises except the last axis
+    for i in range(size_of_tensor_dim - 1):
+        mins = torch.min(mins, 0)[0]
+        maxs = torch.max(maxs, 0)[0]
```

Comment:
maybe you can do something like this to make it more clear: 
1. figure out the axis you need to reduce, by getting all the axis and remove the input axis:
```
axis = list(range(input.dim())
axis.remove(axis)
```
2. min/max over the reduce axis
```
def multi_max(input, axes, keepdim=False):
    '''
    Performs `torch.max` over multiple dimensions of `input`
    '''
    axes = sorted(axes)
    maxed = input
    for axis in reversed(axes):
        maxed, _ = maxed.max(axis, keepdim)
    return maxed
```